tests/integration/preprocessor/_regrid/test_regrid.py

```diff
@@ -81,6 +81,7 @@ def test_regrid__linear_extrapolate_with_mask(self):
         expected.mask = ma.masked
         expected[:, 1, 1] = np.array([1.5, 5.5, 9.5])
         self.assertArrayEqual(result.data, expected)
+        self.assertArrayEqual(result.data.mask, expected.mask)
```

Comment:
`assertArrayEqual` is an instance of `unittest.TestCase`, it's old and should be replaced with the numpy tester but that means we should go though all the tests and that's beyond the scope of this PR

Reply:
Indeed, as @valeriupredoi said, this is based on the older Python [`unittest`](https://docs.python.org/3/library/unittest.html) framework. Pytest is more modern and concise and preferred for new tests.

`assertArrayEqual` is defined here: https://github.com/ESMValGroup/ESMValCore/blob/74d251d828186c4488595382d6ec4ce05f52bd26/tests/__init__.py#L69-L71

The simplest way to improve the tests would probably be to replace this function by:
```python
@wraps(np.testing.assert_array_equal)
def assertArrayEqual(self, a, b, err_msg='', verbose=True):  # noqa: N802
    np.testing.assert_array_equal(a, b, err_msg=err_msg, verbose=verbose)
    if np.ma.isMaskedArray(a) or np.ma.isMaskedArray(b): 
        np.testing.assert_array_equal(a.mask, b.mask, err_msg=err_msg, verbose=verbose)
```

For the other tests, it might also be nicer to simply define a function
```python
def assert_masked_array_equal(a, b):
    np.testing.assert_array_equal(a, b)
    np.testing.assert_array_equal(a.mask, b.mask)
```
and using that instead of `np.testing.assert_array_equal` directly and then having to remember to always test the mask too.

The command `pytest tests/integration/preprocessor/_regrid/test_regrid.py` runs the tests as expected for me.